Watch Extension/AppState.swift

```diff
@@ -49,32 +22,137 @@ struct AppState {
+            self = .communication(newValue)
+        }
+    }
+
+    var savedFyiDialog: SavedFyiDialogAction? {
+        get {
+            guard case let .savedFyiDialog(value) = self else { return nil }
+            return value
+        }
+        set {
+            guard case .savedFyiDialog = self, let newValue = newValue else { return }
+            self = .savedFyiDialog(newValue)
+        }
+    }
+
+    var feeding: FeedingAction? {
+        get {
+            guard case let .feeding(value) = self else { return nil }
+            return value
+        }
+        set {
+            guard case .feeding = self, let newValue = newValue else { return }
+            self = .feeding(newValue)
+        }
+    }
+}
+
+enum SessionAction {
     case loggedIn
     case loggedOut
+}
+
+enum PulseAction {
     case timerPulse
+}
+
+enum CommunicationAction {
     case clearFailedCommunication
+}
+
+enum SavedFyiDialogAction {
     case hideSavedFyiDialog
+}
+
+enum FeedingAction {
     case start(type: FeedingType, side: FeedingSide)
     case stop(Feeding)
     case pause(Feeding)
     case resume(Feeding)
 }
 
-// TODO: see what can be DRY-ed up here
-// swiftlint:disable function_body_length
-func appReducer(value: inout AppState, action: AppAction) {
+let appReducer: (inout AppState, AppAction) -> Void  = combine(
+    pullback(accountStatusReducer, value: \.sessionState, action: \.session),
+    pullback(pulseReducer, value: \.timerPulseCount, action: \.pulse),
+    pullback(communicationReducer, value: \.didCommunicationFail, action: \.communication),
+    pullback(savedFyiDialogReducer, value: \.showSavedFyiDialog, action: \.savedFyiDialog),
+    pullback(feedingReducer, value: \.self, action: \.feeding)
+)
+
+func accountStatusReducer(sessinState: inout SessionState, action: SessionAction) {
     switch action {
     // TODO: can logged out and logged in be combined?
     case .loggedIn:
-        value.session = .loggedIn
+       sessinState = .loggedIn
     case .loggedOut:
-        value.session = .loggedOut
+        sessinState = .loggedOut
+    }
+}
+
+func pulseReducer(timerPulseCount: inout Int, action: PulseAction) {
+    switch action {
     case .timerPulse:
-        value.timerPulseCount += 1
+        timerPulseCount += 1
+    }
+}
+
+// TODO: this should probably be a higher order reducer /w abstraced feeding reducer mutations
+func communicationReducer(didCommunicationFail: inout Bool, action: CommunicationAction) {
+    switch action {
     case .clearFailedCommunication:
-        value.didCommunicationFail = false
+        didCommunicationFail = false
+    }
+}
+
+func savedFyiDialogReducer(showSavedFyiDialog: inout Bool, action: SavedFyiDialogAction) {
+    switch action {
     case .hideSavedFyiDialog:
-        value.showSavedFyiDialog = false
+        showSavedFyiDialog = false
+    }
+}
+
+// TODO: see what can be DRY-ed up here
+func feedingReducer(value: inout AppState, action: FeedingAction) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jasonzurita/BabyPatterns/blob/8654e194a44d1ed4eeee47b0bbd2a871d48433c0/Watch Extension/AppState.swift#L154"></span>Function body should span 40 lines or less excluding comments and whitespace: currently spans 59 lines<br />
<code>function_body_length</code> <code>AppState.swift:154</code></td>
    </tr>
  </tbody>
</table>

Reply:
This should be resolved with higher order reducers / when thinking about DRY-ing up the code.